File: lib/features/about_jci/Domain/entities/BoardRole.dart
class BoardRole{
  final String id;
  final String name;
  final int priority;

  BoardRole({required this.id, required this.name, required this.priority});

  static List<BoardRole> createBoardRoles() {
    return [
      BoardRole(id: '1', name: 'Priority 1', priority: 1),
      BoardRole(id: '2', name: 'Priority 2', priority: 2),
      BoardRole(id: '3', name: 'Priority 3', priority: 3),
    ];
  }
}

File: Server/src/models/Board/BoardRole.ts
import mongoose, { Document, Schema } from "mongoose";

export interface BoardRole extends Document{

    name: string
   priority:number


    
}
export const BoardRoleSchema=new Schema({
  
    name: { type: String, required: true },
    priority:{type:Number,default:0,required:true},

},{
    toJSON:{
        transform(doc,ret){
            delete ret.__v
            delete ret.createdAt
            delete ret.updatedAt
        }
    },
    
    timestamps:true
})
   



const BoardRole=mongoose.model<BoardRole>('BoardRole',BoardRoleSchema)
export { BoardRole };

